src/shared/components/structureViewer/StructureViewerPanel.tsx

```diff
@@ -520,8 +546,98 @@ export default class StructureViewerPanel extends React.Component<IStructureView
     }
 
     private handlePyMolDownload() {
-        const filename = `${this.props.pdbId}_${this.props.chainId}.pml`;
-        fileDownload(this.pyMolScript, filename);
+        if (this.pdbId && this.chainId) {
+            const filename = `${this.pdbId}_${this.chainId}.pml`;
+            fileDownload(this.pyMolScript, filename);
+        }
+    }
+
+    @computed get pdbId()
+    {
+        if (this.pdbChain) {
+            return this.pdbChain.pdbId;
+        }
+        else {
+            return undefined;
+        }
+    }
+
+    @computed get chainId()
+    {
+        if (this.pdbChain) {
+            return this.pdbChain.chain;
+        }
+        else {
+            return undefined;
+        }
+    }
+
+    @computed get pdbChain()
+    {
+        let data = this.props.pdbChainDataStore.sortedFilteredSelectedData;
+
+        if (data.length === 0) {
+            data = this.props.pdbChainDataStore.sortedData;
```

Comment:
Here, I am assuming that we will only sort the chain data once. But if we make the PDB chain table sortable, then it makes things complicated. We don't want the order of chains in the PDB panel to change. Maybe we just don't pass this store to the table component?

On the other hand, maybe it is better to assume that there will always be selected data, and it will always contain one chain and not even worry about the sorted data?